src/Microsoft.AspNetCore.Dispatcher/Patterns/RoutePatternParser.cs

```diff
@@ -469,10 +466,21 @@ private static bool IsValidLiteral(TemplateParserContext context, string literal
             return true;
         }
 
-        private static bool IsInvalidPattern(string routeTemplate)
+        private static string PatternStartsWithSlashOrTildeSlash(string routePattern)
```

Comment:
This should be have a smarter name like `TrimPrefix` or `HandleLeadingSlash`. Based on the naming of this I expect this to return a boolean